### PR TITLE
DOC: add development doc

### LINKED ISSUE #871: DOC: Add doc about installing from source code
Note that the issue tracker is NOT the place for general support.

### LINKED ISSUE #1058: DOC: add how to contribute doc
add documents on how to contribute to this project.

here are some references:

* https://github.com/xorbitsai/xorbits/blob/main/doc/source/development/contributing.rst

### PR DESCRIPTION
**Pull request by Ago327**
add documents(English and Chinese) on how to contribute to this project for issue #1058 and #871
Fixes #1058 
Fixes #871

### CODE REVIEW

**Comment by ChengjieLi28** [comment]
Could you please also translate this to Chinese? That is, add po files for Chinese language.

File: doc/source/index.rst [hunk 1]
```diff
@@ -12,6 +12,7 @@ Welcome to Xinference!
    models/index
    user_guide/index
    examples/index
+   development/index
```
Comment (qinxuye) [inline doc/source/index.rst:15]: The development could be behind reference.

**Review by qinxuye — COMMENTED** [review]
Besides, modify the https://github.com/xorbitsai/inference/blob/eb5311012e0f19f0b8f78ea2da120774450a5ebe/doc/source/conf.py#L78 to 7 to avoid dropdown.

**Comment by qinxuye** [comment]
image

One tip, you can preview the details here to preview the English and Chinese docs.

**Pull request closed without merging**